core/dive.c

```diff
@@ -1836,11 +1849,11 @@ struct sample *add_sample(struct sample *sample, int time, struct divecomputer *
  * that the time in between the dives is at the surface, not some "last
  * sample that happened to be at a depth of 1.2m".
  */
-static void merge_one_sample(struct sample *sample, int time, struct divecomputer *dc)
+static void merge_one_sample(const struct sample *sample, int time, struct divecomputer *dc)
 {
 	int last = dc->samples - 1;
 	if (last >= 0) {
-		static struct sample surface = { .bearing.degrees = -1, .ndl.seconds = -1 };
+		struct sample surface = { .bearing.degrees = -1, .ndl.seconds = -1 };
```

Comment:
I realize that the pointless initialization was there before, but given that these values will be overwritten just four lines below, it does seem silly to keep it.
I'm trying to figure out why this used to be a static variable, but I'm guessing that made sense only in an earlier incarnation of the code?

Reply:
actually, I can't seem to be able to comment on the code that isn't close to the modification (and apparently that's what ended my previous review), but I wonder if the fact that this sample is passed into add_sample() below was the (possibly misguided, need to reed that function again) reason to have the surface sample static
No, thankfully that's not the case, that would have been scary-insane. So yes, long story short, dropping 'static' seems like the right thing to do, and you could also drop the initialization.